stdlib/3/tkinter/__init__.pyi

```diff
@@ -11,7 +12,69 @@ READABLE: Any
 WRITABLE: Any
 EXCEPTION: Any
 
-class Event: ...
+if sys.version_info >= (3, 6):
+    class EventType(str, Enum):
+        Activate: str = ...
+        ButtonPress: str = ...
+        ButtonRelease: str = ...
+        Circulate: str = ...
+        CirculateRequest: str = ...
+        ClientMessage: str = ...
+        Colormap: str = ...
+        Configure: str = ...
+        ConfigureRequest: str = ...
+        Create: str = ...
+        Deactivate: str = ...
+        Destroy: str = ...
+        Enter: str = ...
+        Expose: str = ...
+        FocusIn: str = ...
+        FocusOut: str = ...
+        GraphicsExpose: str = ...
+        Gravity: str = ...
+        KeyPress: str = ...
+        KeyRelease: str = ...
+        Keymap: str = ...
+        Leave: str = ...
+        Map: str = ...
+        MapRequest: str = ...
+        Mapping: str = ...
+        Motion: str = ...
+        MouseWheel: str = ...
+        NoExpose: str = ...
+        Property: str = ...
+        Reparent: str = ...
+        ResizeRequest: str = ...
+        Selection: str = ...
+        SelectionClear: str = ...
+        SelectionRequest: str = ...
+        Unmap: str = ...
+        VirtualEvent: str = ...
+        Visibility: str = ...
+
+class Event:
+    serial: int
+    num: int
+    focus: bool
+    height: int
+    width: int
+    keycode: int
+    state: Union[int, str]
+    time: int
+    x: int
+    y: int
+    x_root: int
+    y_root: int
+    char: str
+    send_event: bool
+    keysym: str
+    keysym_num: int
+    if sys.version_info >= (3, 6):
```

Comment:
There are at least some code paths where `type` and `widget` end up as str, but maybe this doesn't actually happen in practice.

Reply:
Are those just the `??` case? Or are there non-`??` cases where that can happen too?

Reply:
Again, not familiar with the module, but I was checking against: https://github.com/python/cpython/blob/d31b376bc85dded9d059259737b204e544707d07/Lib/tkinter/__init__.py#L1587-L1592 where if some lookups fail, we end up with the (presumably) str argument.
But seems possible this might not really happen in practice.